tools/vgm2la/src/VGMHeader.cpp

```diff
@@ -0,0 +1,189 @@
+// -*- coding: utf-8-unix -*-
+/*
+ * Copyright (c) 2021-2023 Daishi Mori (mori0091)
+ *
+ * This software is released under the MIT License.\n
+ * See https://github.com/mori0091/libmsx/blob/main/LICENSE
+ *
+ * GitHub libmsx project\n
+ * https://github.com/mori0091/libmsx
+ */
+/**
+ * \file VGMHeader.cpp
+ */
+
+#include "VGMHeader.hpp"
+
+void VGMHeader::print() {
+  std::cout << "VGM version : 0x" << std::hex << version << std::dec << std::endl;
+  if (AY8910_clock) {
+    std::cout << "PSG (AY-3-8910) clock : " << AY8910_clock << std::endl;
+  }
+  if (K051649_K052539_clock & (1 << 31)) {
```

Comment:
line 22 ; signed integer overflow for expression "1<<31"
It shall be "1UL << 31".